crates/curp/src/server/conflict/spec_pool_new.rs

```diff
@@ -0,0 +1,129 @@
+use curp_external_api::conflict::SpeculativePool;
+
+use crate::rpc::PoolEntry;
+
+use super::{CommandEntry, ConfChangeEntry, SplitEntry};
+
+/// A speculative pool object
+pub type SpObject<C> = Box<dyn SpeculativePool<Entry = CommandEntry<C>> + Send + 'static>;
+
+/// Union type of `SpeculativePool` objects
+pub(crate) struct SpecPool<C> {
+    /// Command speculative pools
+    command_sps: Vec<SpObject<C>>,
```

Comment:
Why is plural? Do we have more than one speculative pool?

Reply:
Yes, an `SpObject` is created for each type of command, for example, there could be a KV command or an Auth command.